protocol/backend/backend.go

```diff
@@ -228,3 +232,29 @@ func (b *backend) ClearXMRDepositAddress(offerID types.Hash) {
 	defer b.perSwapXMRDepositAddrRWMu.Unlock()
 	delete(b.perSwapXMRDepositAddr, offerID)
 }
+
+// HandleRelayClaimRequest validates and sends the transaction for a relay claim request
+func (b *backend) HandleRelayClaimRequest(request *message.RelayClaimRequest) (*message.RelayClaimResponse, error) {
+	// In the taker relay scenario, the net layer has already validated that we
+	// have an ongoing swap with the requesting peer that uses the passed
+	// offerID, but we have not verified that the claim in the swap matches the
+	// offerID. The backend, with its access to the recovery DB, is in the best
+	// position to perform this check. The remaining validations will be in the
+	// relayer library.
+	if request.OfferID != nil {
+		swapInfo, err := b.recoveryDB.GetContractSwapInfo(*request.OfferID)
```

Comment:
check that we're the xmrtaker for this also as a sanity check? I assume it would be caught in later validations but still

Reply:
The net layer, in the hash of swap connections by offerID, now tracks who initiated the swap connection. In order for us to accept a  taker-specific relay request, the code now checks that the entry exists in the hash, and that the swap entry was created because we (the taker) initiated the connection. I added that today when converting the lock on the net-layer's hash to a RW lock.